hw_3.Functions.py

```diff
@@ -0,0 +1,158 @@
+
+    sheets_for_people = SHEETS_PER_DAY * people
+    sheets_for_14_days = sheets_for_people * DAYS
+
+    if sheets_for_14_days % SHEETS_PER_ROLL != 0:
+        need_roll = (sheets_for_14_days // SHEETS_PER_ROLL) + 1
+    else:
+        need_roll = sheets_for_14_days // SHEETS_PER_ROLL
+    if tp >= need_roll:
+        return print('You have enough toilet paper')
+    else:
+        buy = need_roll - tp
+        return print('You need to buy', buy, 'TP!')
+
+d = {}
+print('Number of people in the household ', end='')
+d["people"] = input()
+print('Number of rolls ', end='')
+d["tp"] = input()
+
+toilet_paper(**d)
+
+
+# 3) Make a function that encrypts a given input with these steps:
+# Input: "apple"
+# Step 1: Reverse the input: "elppa"
+# Step 2: Replace all vowels using the following chart:
+# a => 0
+# e => 1
+# i => 2
+# o => 2
+# u => 3
+# # "1lpp0"
+# Example:
+# encrypt("banana") ➞ "0n0n0baca"
+# encrypt("karaca") ➞ "0c0r0kaca"
+# encrypt("burak") ➞ "k0r3baca"
+# encrypt("alpaca") ➞ "0c0pl0aca"
+def encrypts():
+    # Input: "apple"
+    word = input('Input word: ')
+
+    # Step 1: Reverse the input: "elppa"
+    word1 = ''.join(item[::-1] for item in word.split())
+
+    # Step 2: Replace all vowels using the following chart:
+    d = {'a': 0, 'e': 1, 'i': 2, 'o': 2, 'u': 3}
+    word2 = ''.join(map(str, [d[x] if x in d.keys() else x for x in word1]))
+
+    return print(word2)
+
+encrypts()
+
+# **4)Given a 3x3 matrix of a completed tic-tac-toe game, create a function that returns whether the game is a win
+# for "X", "O", or a "Draw", where "X" and "O" represent themselves on the matrix, and "E" represents an empty spot.
+# Example:
+# tic_tac_toe([
+#     ["X", "O", "X"],
+#     ["O", "X", "O"],
+#     ["O", "X", "X"]
+# ]) ➞ "X"
+#
+# tic_tac_toe([
+#     ["O", "O", "O"],
+#     ["O", "X", "X"],
+#     ["E", "X", "X"]
+# ]) ➞ "O"
+#
+# tic_tac_toe([
+#     ["X", "X", "O"],
+#     ["O", "O", "X"],
+#     ["X", "X", "O"]
+# ]) ➞ "Draw"
+
+def game_tic_tac_toe(matrix):
+    winO = False
+    winX = False
+
+    def determine_the_winner(d):
+        if ''.join(x for x in d) == 'OOO':
+            nonlocal winO
+            winO = True
+            return print('O')
+        elif ''.join(x for x in d) == 'XXX':
+            nonlocal winX
+            winX = True
+            return print('X')
+
+    d1 = []
+    d2 = []
+    for i in range(len(matrix)):
+        d1.append(matrix[i][i])
+        d2.append(matrix[i - 1][-i])
+    determine_the_winner(d1)
+    determine_the_winner(d2)
+
+    # rows
+    for i in range(len(matrix)):
+        determine_the_winner(matrix[i])
+
+    # colms
+    colm0 = []
+    colm1 = []
+    colm2 = []
+    for i in range(len(matrix)):
+        colm0.append(matrix[i][0])
+        colm1.append(matrix[i][1])
+        colm2.append(matrix[i][2])
+    determine_the_winner(colm0)
+    determine_the_winner(colm1)
+    determine_the_winner(colm2)
+
+    if winX!= True and winO != True: return print('Draw')
```

Comment:
"if not winX and not winO:" is more pythonic syntax

Reply:
Thanks. Done